misc/dbt-materialize/dbt/include/materialize/macros/deploy/deploy_init.sql

```diff
@@ -181,7 +183,140 @@
             {% endset %}
             {{ run_query(create_cluster) }}
             {{ set_cluster_ci_tag() }}
+            {{ internal_copy_cluster_grants(schema, deploy_schema) }}
         {% endif %}
     {% endif %}
 {% endfor %}
 {% endmacro %}
+
+{% macro internal_copy_schema_default_privs(from, to) %}
+{% set revoke_initial_defaults %}
+SELECT
+  'ALTER DEFAULT PRIVILEGES ' ||
+  CASE
+    WHEN object_owner = 'PUBLIC' THEN 'FOR ALL ROLES '
+    ELSE 'FOR ROLE ' || object_owner
+  END ||
+  'IN SCHEMA quote_ident({{ to }}) ' ||
+  'REVOKE '    || privilege_type || ' ' ||
+  'ON ' || object_type || 's ' ||
+  'TO ' || grantee || ';'
+FROM mz_internal.mz_show_default_privileges
+WHERE database = current_database() AND schema = lower(trim('{{ from }}'))
+    AND object_owner <> 'none' AND grantee <> 'none'
+{% endset %}
```

Comment:
IMO it's a little fragile to list out the default privileges, because those might be changing out from under you by e.g. Terraform. I think it'll be less fragile to just list the _actual_ privileges that got applied to the schema and revoke those. Yes, technically someone could be messing with those at the same time, but that schema is _actually_ under the control of dbt, where as the default privileges on the database are under someone else's control (Terraform, manual, whatever).

Reply:
That makes sense.